sa/sa.go

```diff
@@ -1328,3 +1297,117 @@ func (ssa *SQLStorageAuthority) GetOrder(ctx context.Context, req *sapb.OrderReq
 
 	return order, nil
 }
+
+func (ssa *SQLStorageAuthority) getAuthorizations(ctx context.Context, table string, status string,
+	registrationID int64, names []string, now time.Time) (map[string]*core.Authorization, error) {
+	if len(names) == 0 {
+		return nil, berrors.InternalServerError("no names received")
+	}
+
+	params := make([]interface{}, len(names))
+	qmarks := make([]string, len(names))
+	for i, name := range names {
+		id := core.AcmeIdentifier{Type: core.IdentifierDNS, Value: name}
+		idJSON, err := json.Marshal(id)
+		if err != nil {
+			return nil, err
+		}
+		params[i] = string(idJSON)
+		qmarks[i] = "?"
+	}
+
+	var auths []*core.Authorization
+	_, err := ssa.dbMap.Select(
+		&auths,
+		fmt.Sprintf(`SELECT %s FROM %s
```

Comment:
Skipped this but totally forgot to comment why: I think this would end up duplicating more code than the abstraction would help in terms of simplifying things. The only difference between the authz and pending authz models are the lock column, which we don't need for this specific case and having specific methods for them then requires a further level of de-abstraction to move from the two model types to the `core.Authorization` type.

While I agree that string interpolation in these SQL queries is pretty gross I think that replacing this with two functions that essentially end up with the exact same code just to avoid it is probably more gross in the long term?

Reply:
Ah, I missed that `selectAuthz` and `selectPendingAuthz` have different return types. You're right, this seems like the least-worst option for now.